Kernel/Tasks/Thread.h

```diff
@@ -1233,6 +1236,7 @@ class Thread
     bool m_is_crashing { false };
     bool m_is_promise_violation_pending { false };
     Atomic<bool> m_have_any_unmasked_pending_signals { false };
+    Atomic<bool> m_was_interrupted { false };
```

Comment:
When setting this variable you use seq_cst (the default memory order) and when reading, consume.

Consume is a weird memory order and I wouldn't recommend you use it, especially since it's scheduled for deprecation in C++26.

Using seq_cst is a bit eh, but I'm also not quite sure what memory order we would need here.

Reply:
It'd be nice to get rid of the seq_cst in the future, but for now it's good enough I guess.